lib/fog/ovirt/requests/compute/v4/list_vm_volumes.rb

```diff
@@ -13,7 +13,12 @@ def list_vm_volumes(vm_id)
               interface = attachment.interface
               attachment_disk.bootable = bootable if attachment_disk.bootable.nil?
               attachment_disk.interface = interface if attachment_disk.interface.nil?
-              attachment_disk.storage_domain = attachment_disk.storage_domains[0].id
+              attachment_disk.storage_domain = begin
+                                                 attachment_disk.storage_domains[0].id || 0
```

Comment:
This assumes there is always a storage domain with id=0.
Is it the correct behavior? I think I would go with `nil`

Reply:
Also we don't have to add a `rescue` statement according to the error described in the issue:
```ruby
attachment_disk.storage_domains&.first&.id
```
This will do the trick of being defensive here.

Reply:
Actually, we just "implemented" what was said is a "good" workaround: https://bugzilla.redhat.com/show_bug.cgi?id=1882404#c9
(better than to have it completely broken)